File: frontend/src/services/exerciseService.js
import api from './api';

export const exerciseService = {
  
  // @GetMapping
  getAllExercises: async (page = 0) => {
    const response = await api.get('/exercises', {
      params: { page }
    });
    return response.data;
  },

  // @GetMapping("/{id}")
  getExerciseById: async (id) => {
    const response = await api.get(`/exercises/${id}`);
    return response.data;
  },

  // @GetMapping("/search")
  searchExercises: async (keyword) => {
    const response = await api.get('/exercises/search', {
      params: { keyword }
    });
    return response.data;
  },

  // @GetMapping("/filter")
  filterExercises: async (filterParams) => {
    // filterParams can include: keyword, category, primaryMuscle,
    // equipment, difficulty, exerciseType, sortBy, direction, page
    const response = await api.get('/exercises/filter', {
      params: filterParams
    });
    return response.data;
  },

  // @GetMapping("/metadata")
  getExerciseMetadata: async () => {
    const response = await api.get('/exercises/metadata');
    return response.data;
  },

  // @GetMapping("/category/{category}/primary-muscles")
  getPrimaryMusclesByCategory: async (category) => {
    const response = await api.get(
      `/exercises/category/${category}/primary-muscles`
    );
    return response.data;
  }

};

File: src/main/java/com/sangamesh/Fitsphere/service/ExerciseService.java
package com.sangamesh.Fitsphere.service;

import com.sangamesh.Fitsphere.dto.exercise.ExerciseCategoryModulesDto;
import com.sangamesh.Fitsphere.dto.exercise.ExerciseDetailDto;
import com.sangamesh.Fitsphere.dto.exercise.ExerciseMetadataDto;
import com.sangamesh.Fitsphere.dto.exercise.ExerciseSummaryDto;
import com.sangamesh.Fitsphere.enums.*;
import org.springframework.data.domain.Page;

import java.util.List;

public interface ExerciseService {

    Page<ExerciseSummaryDto> getAllExercises(int page);

    ExerciseDetailDto getExerciseById(Long id);
    ExerciseCategoryModulesDto getPrimaryMusclesByCategory(Category category);

    List<ExerciseSummaryDto> getExerciseByCategory(Category category);

    List<ExerciseSummaryDto> getExerciseByPrimaryMuscle(Muscle muscle);

    List<ExerciseSummaryDto> searchExercises(String Keyword);

    Page<ExerciseSummaryDto> filterExercises(String keyword,Category category, Muscle primaryMuscle,
                                             Equipment equipment, Difficulty difficulty,
                                             ExerciseType exerciseType,   String sortBy,
                                             String direction, int page);

    ExerciseMetadataDto getExerciseMetadata();
}
